src/views/withNavigation.js

```diff
@@ -0,0 +1,17 @@
+/* @flow */
+
+import React from 'react';
+
+import { ContextWithNavigation } from '../TypeDefinition';
+
+export default function (Component: ReactClass<T>) {
+  const withNavigation = (props: T, { navigation }: ContextWithNavigation) => (
```

Comment:
Can you also set displayName? https://github.com/react-community/react-navigation/blob/master/src/withCachedChildNavigation.js#L25

And the function name should be something else, coz `withNavigation` is the name of the HOC